scrapinghub/client/projects.py

```diff
@@ -121,4 +124,53 @@ def __init__(self, client, projectid):
         self.activity = Activity(_Activity, client, projectid)
         self.collections = Collections(_Collections, client, projectid)
         self.frontiers = Frontiers(_HSFrontier, client, projectid)
-        self.settings = Settings(client._hsclient, projectid)
+        self.settings = Settings(_Settings, client, projectid)
+
+
+class Settings(_MappingProxy):
+    """Class representing job metadata.
+
+    Not a public constructor: use :class:`Project` instance to get a
+    :class:`Settings` instance. See :attr:`Project.settings` attribute.
+
+    Usage::
+
+    - get project settings instance
+
+        >>> project.settings
+        <scrapinghub.client.projects.Settings at 0x10ecf1250>
+
+    - iterate through project settings
+
+        >>> project.settings.iter()
+        <dictionary-itemiterator at 0x10ed11578>
+
+    - list project settings
+
+        >>> project.settings.list()
+        [(u'default_job_units', 2),
+         (u'job_runtime_limit', 20)]
+
+    - get setting value by name
+
+        >>> project.settings.get('default_job_units')
+        2
+
+    - update setting value (some settings are read-only)
+
+        >>> project.settings.set('default_job_units', 2)
+
+    - update multiple settings at once
+
+        >>> project.settings.update({'default_job_units': 1,
+        ...                          'job_runtime_limit': 20})
+
+    - delete project setting by name
+
+        >>> project.settings.delete('job_runtime_limit')
+    """
+    def set(self, key, value):
+        # FIXME drop the method when post-by-key is implemented on server side
+        if not isinstance(key, six.string_types):
+            raise InvalidUsage("key should be a string")
```

Comment:
again, kinda unexpected use of `InvalidUsage`, should be TypeError. What do you think about renaming `InvalidUsage` to `BadRequest` after the 400 response message? Just like it's done for Unauthorized and NotFound?